packages/emitter-framework/src/csharp/components/json-converter/json-converter-resolver.tsx

```diff
@@ -0,0 +1,105 @@
+import { useTsp } from "#core/index.js";
+import { createContext, refkey, useContext, type Children, type Refkey } from "@alloy-js/core";
+import {
+  getTypeName,
+  type DurationKnownEncoding,
+  type EncodeData,
+  type Type,
+} from "@typespec/compiler";
+import { getNullableUnionInnerType } from "../utils/nullable-util.js";
+import { TimeSpanIso8601JsonConverter, TimeSpanSecondsJsonConverter } from "./json-converter.jsx";
+
+interface JsonConverterInfo {
+  refkey: Refkey;
+  name: string;
+  converter: Children;
+}
+
+/**
+ * Help to resolve JsonConverter for a given type with encoding:
+ *   1. Avoid unnecessary duplicate JsonConverter declaration for the same type with same encoding.
+ *   2. Provide resolved JsonConverters to be generated centralized properly as needed.
```

Comment:
As commented and using Context seems the cleanest way to do it, feel free to suggest if there is a better way to do it in EF.